# Comparing `tmp/genofunc-1.1-py3-none-any.whl.zip` & `tmp/genofunc-1.1.1-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,36 +1,36 @@
-Zip file size: 28716 bytes, number of entries: 34
+Zip file size: 28765 bytes, number of entries: 34
 -rw-r--r--  2.0 unx      537 b- defN 22-Mar-25 16:35 genofunc/__init__.py
--rw-r--r--  2.0 unx    16449 b- defN 23-Apr-16 04:21 genofunc/__main__.py
+-rw-r--r--  2.0 unx    16448 b- defN 23-May-10 09:17 genofunc/__main__.py
 -rw-r--r--  2.0 unx     1419 b- defN 22-Mar-25 16:35 genofunc/concatenate_fasta.py
 -rw-r--r--  2.0 unx     2932 b- defN 22-Mar-25 16:35 genofunc/extract_metadata.py
--rw-r--r--  2.0 unx     3780 b- defN 23-Apr-16 07:34 genofunc/feature_extractor.py
+-rw-r--r--  2.0 unx     3781 b- defN 23-May-11 11:44 genofunc/feature_extractor.py
 -rw-r--r--  2.0 unx     4847 b- defN 22-Mar-25 16:35 genofunc/filter_fasta.py
 -rw-r--r--  2.0 unx     1576 b- defN 22-Mar-25 16:35 genofunc/gene_concatenator.py
--rw-r--r--  2.0 unx     6118 b- defN 23-Apr-16 07:27 genofunc/genome_annotator.py
+-rw-r--r--  2.0 unx     6117 b- defN 23-May-10 09:16 genofunc/genome_annotator.py
 -rw-r--r--  2.0 unx     7691 b- defN 22-Jun-10 10:33 genofunc/genome_annotator2.py
 -rw-r--r--  2.0 unx     4677 b- defN 22-Mar-25 16:35 genofunc/group_align.py
 -rw-r--r--  2.0 unx     4106 b- defN 22-Mar-25 16:35 genofunc/merge.py
 -rw-r--r--  2.0 unx     1587 b- defN 22-Mar-25 16:35 genofunc/name_splitter.py
--rw-r--r--  2.0 unx     4739 b- defN 23-Apr-16 04:45 genofunc/reference_matcher.py
+-rw-r--r--  2.0 unx     3924 b- defN 23-May-15 10:21 genofunc/reference_matcher.py
 -rw-r--r--  2.0 unx     1521 b- defN 22-Mar-25 16:35 genofunc/rename_fasta.py
 -rw-r--r--  2.0 unx     2789 b- defN 22-Mar-25 16:35 genofunc/strain_encoder.py
 -rw-r--r--  2.0 unx      650 b- defN 22-Mar-25 16:35 genofunc/utils.py
 -rw-r--r--  2.0 unx      410 b- defN 22-Mar-25 16:35 genofunc/subcommands/__init__.py
 -rw-r--r--  2.0 unx      350 b- defN 22-Mar-25 16:35 genofunc/subcommands/concatenate_fasta.py
 -rw-r--r--  2.0 unx      536 b- defN 22-Mar-25 16:35 genofunc/subcommands/extract_metadata.py
 -rw-r--r--  2.0 unx      418 b- defN 22-Mar-25 16:35 genofunc/subcommands/feature_extractor.py
 -rw-r--r--  2.0 unx      448 b- defN 22-Mar-25 16:35 genofunc/subcommands/filter_fasta.py
 -rw-r--r--  2.0 unx      308 b- defN 22-Mar-25 16:35 genofunc/subcommands/gene_concatenator.py
 -rw-r--r--  2.0 unx      334 b- defN 22-Mar-25 16:35 genofunc/subcommands/genome_annotator.py
 -rw-r--r--  2.0 unx      342 b- defN 22-Mar-25 16:35 genofunc/subcommands/group_align.py
 -rw-r--r--  2.0 unx      325 b- defN 22-Mar-25 16:35 genofunc/subcommands/merge.py
 -rw-r--r--  2.0 unx      373 b- defN 22-Mar-25 16:35 genofunc/subcommands/name_splitter.py
--rw-r--r--  2.0 unx      391 b- defN 23-Apr-16 04:21 genofunc/subcommands/reference_matcher.py
+-rw-r--r--  2.0 unx      391 b- defN 23-May-10 09:17 genofunc/subcommands/reference_matcher.py
 -rw-r--r--  2.0 unx      317 b- defN 22-Mar-25 16:35 genofunc/subcommands/rename_fasta.py
 -rw-r--r--  2.0 unx      397 b- defN 22-Mar-25 16:35 genofunc/subcommands/strain_encoder.py
--rw-r--r--  2.0 unx      570 b- defN 23-Apr-16 09:24 genofunc-1.1.dist-info/METADATA
--rw-r--r--  2.0 unx       92 b- defN 23-Apr-16 09:24 genofunc-1.1.dist-info/WHEEL
--rw-r--r--  2.0 unx       53 b- defN 23-Apr-16 09:24 genofunc-1.1.dist-info/entry_points.txt
--rw-r--r--  2.0 unx        9 b- defN 23-Apr-16 09:24 genofunc-1.1.dist-info/top_level.txt
-?rw-rw-r--  2.0 unx     2909 b- defN 23-Apr-16 09:24 genofunc-1.1.dist-info/RECORD
-34 files, 74000 bytes uncompressed, 24016 bytes compressed:  67.5%
+-rw-r--r--  2.0 unx      572 b- defN 23-May-15 10:28 genofunc-1.1.1.dist-info/METADATA
+-rw-r--r--  2.0 unx       92 b- defN 23-May-15 10:28 genofunc-1.1.1.dist-info/WHEEL
+-rw-r--r--  2.0 unx       53 b- defN 23-May-15 10:28 genofunc-1.1.1.dist-info/entry_points.txt
+-rw-r--r--  2.0 unx        9 b- defN 23-May-15 10:28 genofunc-1.1.1.dist-info/top_level.txt
+?rw-rw-r--  2.0 unx     2919 b- defN 23-May-15 10:28 genofunc-1.1.1.dist-info/RECORD
+34 files, 73196 bytes uncompressed, 24045 bytes compressed:  67.1%
```

## zipnote {}

```diff
@@ -81,23 +81,23 @@
 
 Filename: genofunc/subcommands/rename_fasta.py
 Comment: 
 
 Filename: genofunc/subcommands/strain_encoder.py
 Comment: 
 
-Filename: genofunc-1.1.dist-info/METADATA
+Filename: genofunc-1.1.1.dist-info/METADATA
 Comment: 
 
-Filename: genofunc-1.1.dist-info/WHEEL
+Filename: genofunc-1.1.1.dist-info/WHEEL
 Comment: 
 
-Filename: genofunc-1.1.dist-info/entry_points.txt
+Filename: genofunc-1.1.1.dist-info/entry_points.txt
 Comment: 
 
-Filename: genofunc-1.1.dist-info/top_level.txt
+Filename: genofunc-1.1.1.dist-info/top_level.txt
 Comment: 
 
-Filename: genofunc-1.1.dist-info/RECORD
+Filename: genofunc-1.1.1.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## genofunc/__main__.py

 * *Ordering differences only*

```diff
@@ -109,22 +109,22 @@
         help='Gene regions to be extracted'
     )
     subparser_feature_extractor.add_argument(
         '--strip-gap', dest='strip_gap', action='store_true', required=False,
         help='Strip gap bases within extracted gene regions.'
     )
     subparser_feature_extractor.add_argument(
-        '--filter-coverage', dest='filter_coverage', type=float, metavar='<float>', required=False, default=0.0,
-        help='Minimum gene Non Ambiguious base ratio to sequence length to be filtered (Default: 0.0)'
-    )
-    subparser_feature_extractor.add_argument(
         '--filter-span', dest='filter_span', type=int, metavar='<integer>', required=False, default=0,
         help='Minimum gene sequence length to be filtered (Default: 0)'
     )
     subparser_feature_extractor.add_argument(
+        '--filter-coverage', dest='filter_coverage', type=float, metavar='<float>', required=False, default=0.0,
+        help='Minimum gene Non Ambiguious base ratio to sequence length to be filtered (Default: 0.0)'
+    )
+    subparser_feature_extractor.add_argument(
         '--out-prefix', dest='out_prefix', metavar='<filename>', required=False, default="extracted_",
         help='Output prefix for output sequences (Default: extracted_)'
     )
 
     subparser_feature_extractor.set_defaults(func=genofunc.subcommands.feature_extractor.run)
 
     # _______________________________  filter_fasta  __________________________________#
@@ -378,8 +378,8 @@
     if hasattr(args, "func"):
         args.func(args)
     else:
         parser.print_help()
 
 
 if __name__ == "__main__":
-    main()
+    main()
```

## genofunc/feature_extractor.py

```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env python3
 
 """
 Name: feature_extractor.py
 Author: Xiaoyu Yu
-Date: 01 April 2022
+Date: 10 April 2023
 Description: Extract gene(s) sequences from annotated json file based on user input gene regions.
 
 Options:
     :param in_annotation: Annotated json file containing all sequences (Required)
     :param gene_region: Gene regions to be extracted (Required)
     :param strip_gap: Strip gap bases within gene regions (Default: False)
     :param filter_coverage: Minimum base ratio of non N in gene sequence length to be filtered (Default: 0)
@@ -85,8 +85,8 @@
             new_record = SeqRecord(Seq(seq),id,description="")
             SeqIO.write(new_record, outfile, "fasta-2line")
         close_handle(outfile)
 
     time_ran = dt.datetime.now() - time_start
     print("Time Lapse:", time_ran.total_seconds() / 60, "Minutes")
 
-    close_handle(log_handle)
+    close_handle(log_handle)
```

## genofunc/genome_annotator.py

```diff
@@ -1,33 +1,33 @@
 #!/usr/bin/env python3
 
 """
 Name: genome_annotator.py
 Author: Xiaoyu Yu
-Date: 01 April 2022
+Date: 10 April 2023
 Description: Annotate genomes based on closest reference sequence annotation. 
 
 Options:
     :param raw_fasta: Raw sequences with reference in name tag in fasta format (Required)
     :param reference_sequence: Annotated reference sequences in json format (Required)
-    :param out_annotation: Output list of sequences annotated in json format (Default: referenced.json)
+    :param annotated_json: Output list of sequences annotated in json format (Default: referenced.json)
 
 This file is part of PANGEA HIV project (www.pangea-hiv.org).
 Copyright 2021 Xiaoyu Yu (xiaoyu.yu@ed.ac.uk).
 """
 
 import json
 import sys
 import parasail
 import re
 from Bio import SeqIO
 import datetime as dt
 from genofunc.utils import *
 
-def genome_annotator(raw_fasta,reference_sequence,out_annotation,log_file):
+def genome_annotator(raw_fasta,reference_sequence,annotated_json,log_file):
     def extract_int(base_pos,input_string):
         temp_string = ""
         for j in input_string[base_pos:]:
             if j.isdigit():
                 temp_string += j
             else:
                 break
@@ -146,15 +146,15 @@
                     coordinates[i] = str(gene_coordinate + counter)
                     if temp_value < 0:
                         coordinates[i] = str((gene_coordinate + counter)*(-1))
                     else:
                         coordinates[i] = str(gene_coordinate + counter)
                 reference_dic[id][genes] = "|".join(coordinates)
 
-    with open(out_annotation, 'w') as f:
+    with open(annotated_json, 'w') as f:
         json_dumps_str = json.dumps(reference_dic, indent=4)
         print(json_dumps_str, file=f)
 
     time_ran = dt.datetime.now() - time_start
     print("Time Lapse:", time_ran.total_seconds() / 60, "Minutes")
 
-    close_handle(log_handle)
+    close_handle(log_handle)
```

## genofunc/reference_matcher.py

```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env python3
 
 """
 Name: reference_matcher.py
 Author: Xiaoyu Yu
-Date: 16 April 2023
+Date: 15 May 2023
 Description: Map sequence to the closest reference sequence list based on mini-map2. 
 
 Options:
     :param in_fasta: Raw sequences needed to be referenced to reference list in fasta format (Required)
     :param reference_sequence: Reference list in fasta format (Required)
     :param out_fasta: Output list of sequences referenced (Default: referenced.fasta)
 
@@ -16,19 +16,32 @@
 """
 
 import mappy as mp
 from Bio import SeqIO
 import sys
 from Bio import SeqIO
 from Bio.SeqRecord import SeqRecord
-from Bio.Seq import Seq
 import datetime as dt
 from genofunc.utils import *
 
 def reference_matcher(in_fasta,reference_sequence,virus_type,percentage_match,out_fasta,log_file):
+    def map_to_reference(seq,mismatches,reference_list):
+        matching_length = 0
+        reference_id = ""
+        for hit in reference_list.map(str(seq)):
+            if hit.mlen >= matching_length and hit.NM <= mismatches:
+                matching_length = hit.mlen
+                mismatches = hit.NM
+                reference_id = hit.ctg
+                start_ref = hit.r_st
+                end_ref = hit.r_en
+                cigar_array = hit.cigar
+                log_handle.write("{}\t{}\t{}\t{}\t{}\t{}\t{}\n".format(record.id,hit.ctg,matching_length,mismatches,start_ref,end_ref,cigar_array))
+        return [reference_id,matching_length,mismatches]
+
     time_start = dt.datetime.now()
 
     if not file_check(in_fasta):
         print("Input file does not exist. Please check the file path.")
         sys.exit()
     if not file_check(reference_sequence):
         print("Reference file does not exist. Please check the file path.")
@@ -37,64 +50,38 @@
     outfile = open(out_fasta,"w")
     log_handle = get_log_handle(log_file)
     log_handle.write("Strain\tReference_Sequence\tMatchingBase\tNumberofMismatches\tCigarStart\tCigarArray\n")
 
     reference_list = mp.Aligner(reference_sequence)
 
     for record in SeqIO.parse(in_fasta, "fasta"):
-        matching_length = 0
-        reverse_matching_length = 0
         mismatches = round(float(percentage_match)*len(str(record.seq)),0)
         if virus_type.upper() == "DNA":
-            for hit in reference_list.map(record.seq):
-                if hit.mlen >= matching_length and hit.NM <= mismatches:
-                    matching_length = hit.mlen
-                    mismatches = hit.NM
-                    reference_id = hit.ctg
-                    start_ref = hit.r_st
-                    end_ref = hit.r_en
-                    cigar_array = hit.cigar
-                new_id = "|".join([record.id,str(matching_length),reference_id])
-                new_record = SeqRecord(record.seq,new_id,description="")
-                log_handle.write("{}\t{}\t{}\t{}\t{}\t{}\t{}\n".format(record.id,hit.ctg,matching_length,mismatches,start_ref,end_ref,cigar_array))
-                if matching_length < mismatches:
-                    temp_seq = record.seq.reverse_complement()
-                    for hit in reference_list.map(temp_seq):
-                        if hit.mlen >= matching_length and hit.NM <= mismatches:
-                            reverse_matching_length = hit.mlen
-                            reverse_mismatches = hit.NM
-                            reverse_reference_id = hit.ctg
-                            reverse_start_ref = hit.r_st
-                            reverse_end_ref = hit.r_en
-                            reverse_cigar_array = hit.cigar
-                    if reverse_matching_length > matching_length:
-                        new_id = "|".join([record.id,str(reverse_matching_length),reverse_reference_id])
-                        new_record = SeqRecord(temp_seq,new_id,description="")
-                        log_handle.write("{}\t{}\t{}\t{}\t{}\t{}\t{}\n".format(record.id,hit.ctg,reverse_matching_length,reverse_mismatches,reverse_start_ref,
-                                                                            reverse_end_ref,reverse_cigar_array))                       
-            if matching_length != 0 or reverse_matching_length != 0:
+            mapping_results = map_to_reference(record.seq,mismatches,reference_list)
+            new_id = "|".join([record.id,str(mapping_results[1]),mapping_results[0]])
+            new_record = SeqRecord(record.seq,new_id,description="")
+            if mapping_results[1] < mapping_results[2]:
+                temp_seq = record.seq.reverse_complement()
+                reverse_mapping_results = map_to_reference(temp_seq,mismatches,reference_list)
+                if reverse_mapping_results[1] > mapping_results[1]:
+                    new_id = "|".join([record.id,str(reverse_mapping_results[1]),reverse_mapping_results[0]])
+                    new_record = SeqRecord(temp_seq,new_id,description="")    
+            if mapping_results[0] == "":
+                print("No reference found within the mismatch percentage ratio for " + record.id + ".")
+                continue             
+            if mapping_results[1] != 0 or reverse_mapping_results[1] != 0:
                 SeqIO.write(new_record, outfile, "fasta-2line")
             
         elif virus_type.upper() == "RNA":
-            for hit in reference_list.map(record.seq):
-                if hit.mlen >= matching_length and hit.NM <= mismatches:
-                    matching_length = hit.mlen
-                    mismatches = hit.NM
-                    reference_id = hit.ctg
-                    start_ref = hit.r_st
-                    end_ref = hit.r_en
-                    cigar_array = hit.cigar
-                new_id = "|".join([record.id,str(matching_length),reference_id])
-                new_record = SeqRecord(record.seq,new_id,description="")
-            if matching_length != 0:
+            mapping_results = map_to_reference(record.seq,mismatches,reference_list)
+            new_id = "|".join([record.id,str(mapping_results[1]),mapping_results[0]])
+            new_record = SeqRecord(record.seq,new_id,description="")
+            if mapping_results[1] != 0:
                 SeqIO.write(new_record, outfile, "fasta-2line")
-            log_handle.write("{}\t{}\t{}\t{}\t{}\t{}\t{}\n".format(record.id,hit.ctg,matching_length,mismatches,start_ref,end_ref,cigar_array))
         else:
             print("Virus type not valid, please re-enter correct virus type option (RNA/DNA). Program Exiting")
             sys.exit()
-
-            
     
     time_ran = dt.datetime.now() - time_start
     print("Time Lapse:", time_ran.total_seconds() / 60, "Minutes")
 
-    close_handle(log_handle)
+    close_handle(log_handle)
```

## Comparing `genofunc-1.1.dist-info/METADATA` & `genofunc-1.1.1.dist-info/METADATA`

 * *Files 17% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: genofunc
-Version: 1.1
+Version: 1.1.1
 Summary: UNKNOWN
 Home-page: https://github.com/xiaoyu518/genofunc
 Author: UNKNOWN
 Author-email: UNKNOWN
 License: MIT
 Platform: UNKNOWN
 Classifier: Topic :: Scientific/Engineering :: Bio-Informatics
```

## Comparing `genofunc-1.1.dist-info/RECORD` & `genofunc-1.1.1.dist-info/RECORD`

 * *Files 6% similar despite different names*

```diff
@@ -1,20 +1,20 @@
 genofunc/__init__.py,sha256=GTcLsdLpkTw75d8hjasWQOV_sTsM0cmvWRsY9NCOvco,537
-genofunc/__main__.py,sha256=9KJ-n1PauFFHNewlMJv1ETc7kn1fj07eAZYn1Ltfbmk,16449
+genofunc/__main__.py,sha256=m3vOExraa1TYoRWqfsSUV-pY9EOl7nDtRf1tUyprfuU,16448
 genofunc/concatenate_fasta.py,sha256=kZAxSll72Dbcsi-SOfz8WO0Fk6YUncGn9-HZzYEXGm0,1419
 genofunc/extract_metadata.py,sha256=OhdtJbAEzOyXOBaxf8k2ffuDWpeBnSnzkTHLTn4ngH8,2932
-genofunc/feature_extractor.py,sha256=gXH4A7oMsO95V4VedrJRNaN5SJuE9UM-WJNbJ8DkPv8,3780
+genofunc/feature_extractor.py,sha256=6kotHlO0PxqgV40O_ZI-7wb3UKhPXQ9bQqJb73jyqq8,3781
 genofunc/filter_fasta.py,sha256=d65e2v51m8HRt1MQgfJq03owvCcz2FJ0KDa491AqF2g,4847
 genofunc/gene_concatenator.py,sha256=U0VyT6R4HKRwtH54cb_h8873rlmfXRoriyY_D1dSI3U,1576
-genofunc/genome_annotator.py,sha256=Mdri_W2V4AQz7LVe6JX-L-7L7h9BzTrM8suT1-a9vqM,6118
+genofunc/genome_annotator.py,sha256=f2lnfLavSFx2dAzeyNimB_q6mhCZynyVtkQX_15Etwg,6117
 genofunc/genome_annotator2.py,sha256=4PUGUaZKJYd-DJIPhHcKYvyLVT7fbKMlqAalktKz4tM,7691
 genofunc/group_align.py,sha256=qj0jnLN3v-sE1UjApBDHosGRlfcdBCoNKBrjfcvEErQ,4677
 genofunc/merge.py,sha256=XHb626y6u5plNmcfejRXmgzO3D17pW7J4nC96PnvYUk,4106
 genofunc/name_splitter.py,sha256=LMvmfMCSGSHKrShR-bIim5nALq4Q1dKV30X4xpCnIH4,1587
-genofunc/reference_matcher.py,sha256=jaKaEvunWQbSdar6Zqbif8NFbuTf6wo_AholFsCqQfQ,4739
+genofunc/reference_matcher.py,sha256=FnhCDGcDiFXK5ZGETVw5y3-Dvs4d4z2ZdkUzYSAq9yQ,3924
 genofunc/rename_fasta.py,sha256=RpIto02NkmdxnpcbPL2rlw69MZ6jMm4Lrx4aq3Q08bA,1521
 genofunc/strain_encoder.py,sha256=tCwzxo5V9Y9_JHApxfL0RX8wg0epgsHk4yBDbX_N_Jc,2789
 genofunc/utils.py,sha256=K0KHBLHCTm9Kgv07IpSllSfHyNzjKBOgPIO7ZYdl4Ng,650
 genofunc/subcommands/__init__.py,sha256=ULmKBq-2JqrAJrh0LKy6yDG_7a4GeTmE7sXjPEmQOHM,410
 genofunc/subcommands/concatenate_fasta.py,sha256=3JD1N14vgcbbAWYKRi725x3OwaI7DLUimbd6_WahGoc,350
 genofunc/subcommands/extract_metadata.py,sha256=ITzNfDsG2WyPhSX1P_SGy6Z57tsJLAjg6O9vIuHp7BM,536
 genofunc/subcommands/feature_extractor.py,sha256=NBTwaasbkeKU2bo3wFqENSK_mwuiO5NYuSIKzVAl8h0,418
@@ -23,12 +23,12 @@
 genofunc/subcommands/genome_annotator.py,sha256=VNF7r9-mtnnqxGJBxnImBmUoJK6hmFBrnJzuCzEw3NQ,334
 genofunc/subcommands/group_align.py,sha256=4dERT7cdM_pgHANrE1WMpBPF03sn11rk0csETmrIU5A,342
 genofunc/subcommands/merge.py,sha256=uTz_oVJ1Tzd958jttdPo8ka3jWIKc0dt4quEuwu2VIw,325
 genofunc/subcommands/name_splitter.py,sha256=s3SMtbRsODSx0RuK6wcByeUJfpPgYLfdMqI1nMJaJYs,373
 genofunc/subcommands/reference_matcher.py,sha256=ArmUvL6JFD8NV2gBMdZrNDnQ4Szby0QLSXKPEQ1oO9w,391
 genofunc/subcommands/rename_fasta.py,sha256=TACml7SbX7ZRRVbJoGNW49I6HdjBxAwj3TVi3N2ClsM,317
 genofunc/subcommands/strain_encoder.py,sha256=wBtrfxFMERiDAEtAziu145M1QChp6prDpGi9RhA5MSk,397
-genofunc-1.1.dist-info/METADATA,sha256=jMEc-XhVzKgECVJIxsvykBY3Pctc_iFm_E1LQFkkjyk,570
-genofunc-1.1.dist-info/WHEEL,sha256=p46_5Uhzqz6AzeSosiOnxK-zmFja1i22CrQCjmYe8ec,92
-genofunc-1.1.dist-info/entry_points.txt,sha256=9xT6cQgMyyC1ktKvIEtsvI1NFjO_W9iloZ9NbIcPw6Q,53
-genofunc-1.1.dist-info/top_level.txt,sha256=rdj71U7IfsnpHsF93VlZG0FTDrPHDxaua4sx_F5yYUI,9
-genofunc-1.1.dist-info/RECORD,,
+genofunc-1.1.1.dist-info/METADATA,sha256=dWaTx7q1VeiFog8PbvLvdwM5HqiYAggU7NaI_h2zXjc,572
+genofunc-1.1.1.dist-info/WHEEL,sha256=U88EhGIw8Sj2_phqajeu_EAi3RAo8-C6zV3REsWbWbs,92
+genofunc-1.1.1.dist-info/entry_points.txt,sha256=9xT6cQgMyyC1ktKvIEtsvI1NFjO_W9iloZ9NbIcPw6Q,53
+genofunc-1.1.1.dist-info/top_level.txt,sha256=rdj71U7IfsnpHsF93VlZG0FTDrPHDxaua4sx_F5yYUI,9
+genofunc-1.1.1.dist-info/RECORD,,
```

